Upgrade to Thrift v0.14

spladug opened the pull request:
Thrift 0.14 is now published as `thrift-unofficial` via https://github.com/reddit/thrift-python/

This allows us to upgrade.

fishy reviewed (approved):
the name sounds... very... unofficial :badpokerface:

spladug committed:
Upgrade to Thrift v0.14.1

0.14.0 had a bug where the constructor of exception types with zero
  fields would crash on instantiation.
